metadata-ingestion/src/datahub/telemetry/telemetry.py

```diff
@@ -183,37 +181,61 @@ def with_telemetry(func: Callable[..., T]) -> Callable[..., T]:
     @wraps(func)
     def wrapper(*args: Any, **kwargs: Any) -> Any:
 
-        action = f"function:{func.__module__}.{func.__name__}"
+        function = f"{func.__module__}.{func.__name__}"
 
         telemetry_instance.init_tracking()
-        telemetry_instance.ping(action)
+        telemetry_instance.ping(
+            "function-call", {"function": function, "status": "start"}
+        )
         try:
             res = func(*args, **kwargs)
-            telemetry_instance.ping(f"{action}:result", {"status": "completed"})
-            return res
-        # Catch general exceptions
-        except Exception as e:
             telemetry_instance.ping(
-                f"{action}:result", {"status": "error", "error": get_full_class_name(e)}
+                "function-call",
+                {"function": function, "status": "completed"},
             )
-            raise e
+            return res
         # System exits (used in ingestion and Docker commands) are not caught by the exception handler,
         # so we need to catch them here.
         except SystemExit as e:
             # Forward successful exits
             if e.code == 0:
```

Comment:
Should we check against `None` as well (see [documentation](https://docs.python.org/3/library/exceptions.html?highlight=systemexit#SystemExit.code))?